napari/_app/actions/_layer_actions.py

```diff
@@ -0,0 +1,161 @@
+"""This module contains actions (functions) that operate on layers.
+
+Among other potential uses, these will populate the menu when you right-click
+on a layer in the LayerList.
+"""
+
+from __future__ import annotations
+
+from functools import partial
+from typing import TYPE_CHECKING, List
+
+from app_model.types import Action
+
+from ...layers import _layer_actions
+from ...utils.translations import trans
+from .._menus import MenuGroup, MenuId
+from ..context import LayerListContextKeys as LLCK
+
+if TYPE_CHECKING:
+    from app_model.types import MenuRuleDict
+
+
+LAYERCTX_SPLITMERGE: MenuRuleDict = {
+    'id': MenuId.LAYERLIST_CONTEXT,
+    'group': MenuGroup.LAYERLIST_CONTEXT.SPLIT_MERGE,
+}
+LAYERCTX_CONVERSION: MenuRuleDict = {
+    'id': MenuId.LAYERLIST_CONTEXT,
+    'group': MenuGroup.LAYERLIST_CONTEXT.CONVERSION,
+}
+LAYERCTX_LINK: MenuRuleDict = {
+    'id': MenuId.LAYERLIST_CONTEXT,
+    'group': MenuGroup.LAYERLIST_CONTEXT.LINK,
+}
+
+_only_labels = LLCK.num_selected_labels_layers == LLCK.num_selected_layers
+
+# sourcery skip: for-append-to-extend
+LAYER_ACTIONS: List[Action] = [
+    Action(
+        id='napari:layers:duplicate_layer',
+        title=trans._('Duplicate Layer'),
+        callback='napari.layers._layer_actions:_duplicate_layer',
```

Comment:
I have worries about such a definition of callback. It will be not handled by any refactor/search usage tool. 

I know that following version may look a little worst, but will be supported by existing tools for code development:

```suggestion
        callback=f'{_duplicate_layer.__module__}:{_duplicate_layer.__name__}',
```

Reply:
yeah, this was partly experimentation too.  I think these should really be direct pointers, not strings (which is also a supported type for callback).  If we have our imports setup correctly, the app should sit at the top and be able to import anything.